test/base/test_decomp.jl

```diff
@@ -49,6 +156,219 @@ using ITensors, LinearAlgebra, Test
+    @test norm(dense(Q * dag(prime(Q, q))) - δ(Float64, q, q')) ≈ 0.0 atol = 1e-13
+  end
+
+  @testset "QR/RQ dense with positive R" begin
+    l = Index(5, "l")
+    s = Index(2, "s")
+    r = Index(10, "r")
+    A = randomITensor(l, s, s', r)
+    Q, R, q = qr(A, l, s, s'; positive=true)
+    @test min(diag(R)...) > 0.0
+    @test A ≈ Q * R atol = 1e-13
+    @test Q * dag(prime(Q, q)) ≈ δ(Float64, q, q') atol = 1e-13
+    R, Q, q = ITensors.rq(A, r; positive=true)
+    @test min(diag(R)...) > 0.0
+    @test A ≈ Q * R atol = 1e-13
+    @test Q * dag(prime(Q, q)) ≈ δ(Float64, q, q') atol = 1e-13
+  end
+
+  @testset "QR/RQ block sparse with positive R" begin
+    l = dag(Index(QN("Sz", 0) => 3; tags="l"))
+    s = Index(QN("Sz", -1) => 1, QN("Sz", 1) => 1; tags="s")
+    r = Index(QN("Sz", 0) => 3; tags="r")
+    A = randomITensor(l, s, dag(s'), r)
+    Q, R, q = qr(A, l, s, s'; positive=true)
+    @test min(diag(R)...) > 0.0
+    @test A ≈ Q * R atol = 1e-13
+    R, Q, q = ITensors.rq(A, r; positive=true)
+    @test min(diag(R)...) > 0.0
+    @test A ≈ Q * R atol = 1e-13
+  end
+
+  test_combos = [(make_Heisenberg_AutoMPO, "S=1/2"), (make_Hubbard_AutoMPO, "Electron")]
+
+  @testset "QR MPO tensors with complex block structures, H=$(test_combo[1])" for test_combo in
+                                                                                  test_combos
+    N, NNN = 10, 2 #10 lattice site, up 7th neight interactions
+    sites = siteinds(test_combo[2], N; conserve_qns=true)
+    H = test_combo[1](sites, NNN)
+    for n in 5:5
+      W = H[n]
+      @test flux(W) == QN("Sz", 0)
+      ilr = filterinds(W; tags="l=$n")[1]
+      ilq = noncommoninds(W, ilr)
+      Q, R, q = qr(W, ilq)
+      @test flux(Q) == QN("Sz", 0) #qr should move all flux on W (0 in this case) onto R
+      @test flux(R) == QN("Sz", 0) #this effectively removes all flux between Q and R in thie case.
+      @test W ≈ Q * R atol = 1e-13
+      # blocksparse - diag is not supported so we must convert Q*Q_dagger to dense.
+      # Also fails with error in permutedims so below we use norm(a-b)≈ 0.0 instead.
+      # @test dense(Q*dag(prime(Q, q))) ≈ δ(Float64, q, q') atol = 1e-13
+      @test norm(dense(Q * dag(prime(Q, q))) - δ(Float64, q, q')) ≈ 0.0 atol = 1e-13
+
+      R, Q, q = ITensors.rq(W, ilr)
+      @test flux(Q) == QN("Sz", 0)
+      @test flux(R) == QN("Sz", 0)
+      @test W ≈ Q * R atol = 1e-13
+      @test norm(dense(Q * dag(prime(Q, q))) - δ(Float64, q, q')) ≈ 0.0 atol = 1e-13
+
+      Q, L, q = ITensors.ql(W, ilq)
+      @test flux(Q) == QN("Sz", 0)
+      @test flux(L) == QN("Sz", 0)
+      @test W ≈ Q * L atol = 1e-13
+      @test norm(dense(Q * dag(prime(Q, q))) - δ(Float64, q, q')) ≈ 0.0 atol = 1e-13
+
+      L, Q, q = ITensors.lq(W, ilr)
+      @test flux(Q) == QN("Sz", 0)
+      @test flux(L) == QN("Sz", 0)
+      @test W ≈ Q * L atol = 1e-13
+      @test norm(dense(Q * dag(prime(Q, q))) - δ(Float64, q, q')) ≈ 0.0 atol = 1e-13
+    end
+  end
```

Comment:
Could you move tests that make use of MPS/MPO types into `test/ITensorLegacyMPS`? Relatedly, do you feel like the other tests properly cover the basic functionality, and these are just "extra"?

There are two reasons:
1. Divide the tests into a more "minimal" test set just depending on ITensors which are run by default for all PRs, while other ones are triggered manually or on the `main` branch, to minimize CI testing time on PRs.
2. In preparation for moving the MPS/MPO code into a separate package and deprecating in favor of the new tree tensor network type in https://github.com/mtfishman/ITensorNetworks.jl.

Reply:
No problem moving them.  Yes they can probably be considered "extra" .  I wanted to verify that 1) complicated block structures do not trigger a problem (some combiner effect?), 2) Fermions in the Hubbard model do not trigger something.

Can you think of a reason to have we should have a fermion operator in the core QR tests? If not then they are most likely "extra"

Reply:
In test/ITensorLegacyMPS/base/test_qnmpo.jl or a separate file like test_qnmpo_decomp.jl ?

Reply:
`test_qnmpo.jl` is fine.

Reply:
Done